PluginCommandManager.fs

```diff
@@ -1,50 +1,60 @@
 ﻿namespace DalamudPluginProjectTemplateFSharp
 
 open System
-open System.Collections.Generic
-open System.Linq
 open System.Reflection
 open Dalamud.Game.Command
 open DalamudPluginProjectTemplateFSharp.Attributes
 
-type private LoadedCommandInfo = (string*CommandInfo)
+type private LoadedCommandInfo = (string * CommandInfo)
 
-[<AllowNullLiteral>]
-type PluginCommandManager<'THost>(host : 'THost, commandManager : CommandManager) as this =
-    let GetCommandInfoTuple(method : MethodInfo) : IEnumerable<LoadedCommandInfo> =
-        let handlerDelegate : CommandInfo.HandlerDelegate = downcast Delegate.CreateDelegate(typeof<CommandInfo.HandlerDelegate>, host, method)
+type PluginCommandManager<'THost>(host: 'THost, commandManager: CommandManager) =
+    let commands =
+        host
+            .GetType()
+            .GetMethods(
+                BindingFlags.NonPublic
+                ||| BindingFlags.Public
+                ||| BindingFlags.Static
+                ||| BindingFlags.Instance
+            )
+
+    let getCommandInfoTuple (method: MethodInfo) : LoadedCommandInfo list =
+        let handlerDelegate: CommandInfo.HandlerDelegate =
+            downcast Delegate.CreateDelegate(typeof<CommandInfo.HandlerDelegate>, host, method)
 
         let command = handlerDelegate.Method.GetCustomAttribute<CommandAttribute>()
         let aliases = handlerDelegate.Method.GetCustomAttribute<AliasesAttribute>()
         let helpMessage = handlerDelegate.Method.GetCustomAttribute<HelpMessageAttribute>()
-        let doNotShowInHelp = handlerDelegate.Method.GetCustomAttribute<DoNotShowInHelpAttribute>()
+
+        let doNotShowInHelp =
+            handlerDelegate.Method.GetCustomAttribute<DoNotShowInHelpAttribute>()
 
         let commandInfo = CommandInfo(handlerDelegate)
-        commandInfo.ShowInHelp <- doNotShowInHelp <> null
-        if (helpMessage <> null && helpMessage.HelpMessage <> null) then
+        commandInfo.ShowInHelp <- not (isNull doNotShowInHelp)
+
+        if (not (isNull helpMessage)
+            && not (isNull helpMessage.HelpMessage)) then
             commandInfo.HelpMessage <- helpMessage.HelpMessage
         else
             commandInfo.HelpMessage <- ""
 
-        let commandInfoTuples = List<LoadedCommandInfo>()
-        commandInfoTuples.Add((command.Command, commandInfo))
-
-        upcast commandInfoTuples
+        (command.Command, commandInfo)
+        :: if isNull aliases then
+               []
+           else
+               [ for alias in aliases.Aliases -> (alias, commandInfo) ]
```

Comment:
The handling of aliases was actually completely missing here; I ported the code from the C# version of your template.